build.sbt

```diff
@@ -251,9 +259,36 @@ lazy val scoverageSettings = {
   coverageExcludedPackages := ".+Benchmark.*"
 }
 
+/** Used for publishing `extras`, `api` and `python` jars. Main Spark Job Server assembly is published
+  * as always. */
 lazy val publishSettings = Seq(
+  autoScalaLibrary := false,
+  credentials += Credentials(Path.userHome / ".sbt" / ".credentials"),
+  publishMavenStyle := true,
+  publishTo := Some(sys.env("MVN_PUBLISH_REPO") at sys.env("MVN_PUBLISH_URL")),
   licenses += ("Apache-2.0", url("http://choosealicense.com/licenses/apache/")),
-  bintrayOrganization := Some("spark-jobserver")
+  pomIncludeRepository := { _ => false },
+  /** Since users are encouraged to use dse-spark-dependencies, which provides most of the needed
+    * dependencies, we remove most of the Spark Job Server deps here. Provided, test and blacklisted
+    * deps are removed from resulting poms. */
+  pomPostProcess := { (node: XmlNode) =>
+    new RuleTransformer(new RewriteRule {
+      val pomDependencyBlacklist = Seq("job-server_", "joda-convert", "joda-time")
+      val emptyElement = Text("")
+
+      def hasTestOrProvidedScope(e: Elem): Boolean = e.child.exists(child =>
+        child.label == "scope" && (child.text == "provided" || child.text == "test"))
+
+      def isBlacklisted(e: Elem): Boolean = e.child.exists(child =>
+        child.label == "artifactId" && pomDependencyBlacklist.exists(child.text.startsWith))
+
+      override def transform(node: XmlNode): XmlNodeSeq = node match {
+        case e: Elem if e.label == "dependency" && (hasTestOrProvidedScope(e) || isBlacklisted(e)) =>
+          emptyElement
```

Comment:
Why an empty element here instead of just nothing? We could pass an empty Seq[Node]?